libflux/flux-core/src/semantic/types.rs

```diff
@@ -94,13 +95,19 @@ impl Substitutable for PolyType {
 }
 
 impl MaxTvar for [Tvar] {
-    fn max_tvar(&self) -> Tvar {
-        self.iter().max().cloned().unwrap_or(Tvar(0))
+    fn max_tvar(&self) -> Option<Tvar> {
```

Comment:
Since `None < Some()` all comparisons still work the same, it is just that we distinguish `0` being max and no variables (otherwise the "bug" path triggers in tests).